File: MODUL 6/PRAK604-2310817210021-RAKHMAD FAUZHAN RAMADHAN.py
def dekode_pesan(kode, pesan_diterima):
    pesan_terdekripsi = ""
    hitung_bintang = 0
    hitung_pagar = 0

    for i in range(min(len(kode), len(pesan_diterima))):
        if kode[i] == pesan_diterima[i]:
            pesan_terdekripsi += "*"
            hitung_bintang += 1
        else:
            pesan_terdekripsi += "#"
            hitung_pagar += 1

    return pesan_terdekripsi, hitung_bintang, hitung_pagar

kode = input().strip()
pesan_diterima = input().strip()

pesan_terdekripsi, hitung_bintang, hitung_pagar = dekode_pesan(kode, pesan_diterima)

if len(kode) != len(pesan_diterima):
    print("Panjang kalimat berbeda, pesan palsu")
else:
    print(pesan_terdekripsi)
    print(f"* = {hitung_bintang}")
    print(f"# = {hitung_pagar}")

    if hitung_bintang >= hitung_pagar:
        print("Pesan Asli")
    else:
        print("Pesan Palsu")

File: MODUL 6/PRAK604-2310817210021-RAKHMAD FAUZHAN RAMADHAN.c
#include <stdio.h>
#include <string.h>

void dekode_pesan(char kode[], char pesan_diterima[], char pesan_terdekripsi[], int *hitung_bintang, int *hitung_pagar) {
    for (int i = 0; i < strlen(kode) && i < strlen(pesan_diterima); i++) {
        if (kode[i] == pesan_diterima[i]) {
            pesan_terdekripsi[i] = '*';
            (*hitung_bintang)++;
        } else {
            pesan_terdekripsi[i] = '#';
            (*hitung_pagar)++;
        }
    }
}

int main() {
    char kode[1000], pesan_diterima[1000], pesan_terdekripsi[1000];
    int hitung_bintang = 0, hitung_pagar = 0;

    fgets(kode, sizeof(kode), stdin);
    fgets(pesan_diterima, sizeof(pesan_diterima), stdin);

    kode[strcspn(kode, "\n")] = '\0';
    pesan_diterima[strcspn(pesan_diterima, "\n")] = '\0';

    dekode_pesan(kode, pesan_diterima, pesan_terdekripsi, &hitung_bintang, &hitung_pagar);

    if (strlen(kode) != strlen(pesan_diterima)) {
        printf("Panjang kalimat berbeda, pesan palsu\n");
    } else {
        printf("%s\n", pesan_terdekripsi);
        printf("* = %d\n", hitung_bintang);
        printf("# = %d\n", hitung_pagar);

        if (hitung_bintang >= hitung_pagar) {
            printf("Pesan Asli\n");
        } else {
            printf("Pesan Palsu\n");
        }
    }

    return 0;
}

